src/main/java/com/google/sps/data/PlayerDatabase.java

```
@@ -99,6 +100,17 @@ public String getEntityID() throws LoggedOutException {
     return id;
   }
 
+  // get experience
+  public int getEntityExperience() throws LoggedOutException {
+    String experienceString =
+        getCurrentPlayerEntity().getProperty(EXPERIENCE_QUERY_STRING).toString();
+    int experience = 0;
+    if (experienceString.matches("\\d+")) {
+      experience = Integer.parseInt(experienceString);
```

Why not just parse it and ignore an exception if you can't parse it?